### PR TITLE
enhanced code of user.py with dict.get()

### LINKED ISSUE #1065: Improvement: Refactor code to use dict.get() method
## Description
As a developer,
I need code to be consistent and error-free.

## Mocks
https://github.com/anitab-org/mentorship-backend/blob/d52d77706aad4a486f2417419bb4c4c6d2ace3a6/app/api/dao/user.py#L42-L46
For example this can be replaced by the get method i.e. `data.get("KEY", "DEFAULT_VALUE")` to avoid KeyError

### Update [Required]
- [ ] Use get method in all code to avoid the raise of KeyError
- [ ] Handle exceptions for default value

## Definition of Done
- [ ] All of the required items are completed.
- [ ] Approval by 1 mentor.

## Estimation
2 hours

### PR DESCRIPTION
**Pull request by aakankshaagr**
### Description

Fixes #1065 

### Type of Change:

- Code
- Quality Assurance

### How Has This Been Tested?

Tested on my local pc

### Checklist:

- [ ] My PR follows the style guidelines of this project
- [ ] I have performed a self-review of my own code or materials

**Code/Quality Assurance Only**

- [ ] My changes generate no new warnings 
- [ ] New and existing unit tests pass locally with my changes
- [ ] Any dependent changes have been published in downstream modules

### CODE REVIEW

**Comment by vj-codes** [comment]
@aakankshaagr the build check is failing , can you please look into it ?

**Comment by aakankshaagr** [comment]
sure @vj-codes !! I looked into tests and found that the test case is failing bcz test is to check " " to be treated as none and I guess it's happening bcz of use of dict.get() as it returns none only when the key is absent. If the key is present it will return whatever value user has given.

image

Here there is an error related to timestamp. I don't know how to remove this error.

image

So tests are failing bcz there is an error and a test fail.

**Comment by aakankshaagr** [comment]
image
I tried using `dict.get("key",None) or None` to save variable with none if any false values are given as input but still test is failing.

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
Hi @aakankshaagr thanks for this
There are some more places where this method can be used, like from line 42-47 (mentioned in issue too). Kindly look into the same

**Comment by aakankshaagr** [comment]
Hi @devkapilbansal thanks for helping me out. I thought earlier that its not needed there but I guess using dict.get() there initializes empty values with None. I missed this point earlier

**Comment by aakankshaagr** [comment]
I made the changes but still tests are failing

**Comment by devkapilbansal** [comment]
> I made the changes but still tests are failing

First of all, please update your code with current develop

**Comment by aakankshaagr** [comment]
@devkapilbansal @codesankalp  please check if any changes are required

File: app/api/dao/user.py [hunk 1]
```diff
@@ -244,73 +244,46 @@ def update_user_profile(user_id: int, data: Dict[str, str]):
             user.username = username
 
         if "name" in data and data["name"]:
```
Comment (devkapilbansal) [inline app/api/dao/user.py]: We don't need these if conditions too

File: app/api/dao/user.py [hunk 2]
```diff
@@ -244,73 +244,46 @@ def update_user_profile(user_id: int, data: Dict[str, str]):
...
-            user.name = data["name"]
+            user.name = data.get("name", None) or None
 
         if "bio" in data:
```
Comment (devkapilbansal) [inline app/api/dao/user.py]: Same as above similar for other cases as well

File: app/api/dao/user.py [hunk 3]
```diff
@@ -332,8 +305,8 @@ def change_password(user_id: int, data: Dict[str, str]):
 
         """
 
-        current_password = data["current_password"]
-        new_password = data["new_password"]
+        current_password = data.get("current_password", None)
+        new_password = data.get("new_password", None)
```
Comment (devkapilbansal) [inline app/api/dao/user.py]: None can be removed here as it is the default value returned
```suggestion
        current_password = data.get("current_password")
        new_password = data.get("new_password")
```

**Review by devkapilbansal — CHANGES_REQUESTED** [review]
Hi @aakankshaagr please rebase your branch to current develop. j
Also, there are some minor changes as suggested below :point_down:

**Comment by devkapilbansal** [comment]
Hi @aakankshaagr why this PR is closed??

**Comment by aakankshaagr** [comment]
Hi @devkapilbansal earlier my branch wasn't updated with current develop even though I haven't forgotten to do a git pull to update my codebase and here I needed to rebase my branch to current develop. 
To simplify my work I merged all my commits of this issue into one commit and then rebased that commit to the current develop and made some changes to the code.
While trying to push all that changes, I can't find my changes here in this PR so I created a new PR.

**Pull request closed without merging**